### PR TITLE
Update gazebo rrbot example

### PR DESCRIPTION
**Pull request by haider8645**
Hello,

I noticed that the example of using the rrbot in gazebo together with the example publisher needed some updates. After these changes, the controller is spawned after you load the rrbot_system_position_only_gazebo.launch.py

Best,
Haider

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -286,7 +286,18 @@ Available launch-file options:
 
 ### Example 2: "Industrial Robots with only one interface" (Gazebo simulation)
 
-- **TBA**
+- Launch file: ros2_control_demo_bringup rrbot_system_position_only_gazebo.launch.py
```
Comment (destogl) [inline README.md:289]: This is great! Thanks! Could you also reference "controllers.yaml" and URDF used in that example. See how this is done for other examples.

File: README.md [hunk 2]
```diff
@@ -286,7 +286,18 @@ Available launch-file options:
...
+- Command interfaces:
+  - joint1/position
+  - joint2/position
+- State interfaces:
+  - joint1/position
+  - joint2/position
+
+Available controllers:
+  - `joint_state_broadcaster[joint_state_broadcaster/JointStateBroadcaster] active`
+  - `joint_trajectory_controller[joint_trajectory_controller/JointTrajectoryController] active`
```
Comment (destogl) [inline README.md:301]: Do se need here some additional steps how to run example or is it self-explanatory? How hard was for you to understand everything?

File: ros2_control_demo_bringup/config/rrbot_gazebo_joint_trajectory_controller.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+controller_manager:
```
Comment (destogl) [inline ros2_control_demo_bringup/config/rrbot_gazebo_joint_trajectory_controller.yaml:1]: When we are already working on this example. Would it make sense to add a forwarding controller too? So one can switch controllers is needed? What do you think?

**Review by destogl — COMMENTED** [review]
Looking good! Just a few comments to clarify the situation.

File: README.md (hunk 2 above)
Comment (haider8645) [reply README.md:301]: I add the necessary steps to get it started in the readme and no more steps are needed except for the changes in this pull request.

Regarding your question about difficulty in understanding:

I think that since multiple people contribute so differences in naming convention, file structure/layouts results as a consequence. Like for example

1) Launch File Layout: What I found confusing was different layouts for the files used in the different packages.  I think that once a layout for the launch files can be agreed upon then all the launch files in the example can be refactored to a common layout so that the learner can focus only on the part which changed in different launch files. I made an [example layout](https://github.com/haider8645/ros2_launch_tutorial/blob/master/launch/ros2.launch.py) by combing different layouts  to implement some advanced logic (if needed) in the launch file using the context. This will help use the full potential of the launch file module capabilities. 

2) There were/are some naming discrepancies which I will try to rectify in further pull requests. For example the changes you saw in PR#190

3) I think helper comments can be added to the configs and specially the gazebo files for people who just start in ROS2. I guess that the assumption while creating this package docu was that the person has some ROS1 knowledge about gazebo ros, gazebo_ros_control etc. but since many people start now with ROS2, they might find it hard to follow the complexity. The complexity because you need the name joint names in the gazebo xacro, controller yml, and urdf.. so it can be hard to follow it for new comers. 

PS: I will be working with this package so I will try to create new PR's to enhance the docu and understandability as a user/learner myself.

File: ros2_control_demo_bringup/config/rrbot_gazebo_joint_trajectory_controller.yaml (hunk 3 above)
Comment (haider8645) [reply ros2_control_demo_bringup/config/rrbot_gazebo_joint_trajectory_controller.yaml:1]: Yes that sounds good. I will do that! I think that it will make the difference between the two controllers clearer for the new learners.

Just a question for my understanding: If the user switches to the forwarding controller then there will be snapping movement in between the 4 goal positions right? Since it will forward the commanded position to the joint directly without any control loop. And the joint_traj_controller has the control loop and it keeps the error between commanded and current state so it is a more smoother motion. Do I understand it right?

File: ros2_control_demo_bringup/config/rrbot_joint_trajectory_publisher.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 publisher_joint_trajectory_controller:
   ros__parameters:
 
-    controller_name: "position_trajectory_controller"
+    controller_name: "joint_trajectory_controller"
```
Comment (haider8645) [inline ros2_control_demo_bringup/config/rrbot_joint_trajectory_publisher.yaml:4]: I will change the name of this controller to joint_trajectory_position_controller like in #190

File: README.md (hunk 1 above)
Comment (haider8645) [reply README.md:289]: @destogl I will continue on this PR after the #190 is merged because then the naming can be aligned for the joint_trajectory_position_controller.

File: ros2_control_demo_description/rrbot_description/gazebo/rrbot.gazebo.xacro [hunk 5]
```diff
@@ -10,7 +10,7 @@ https://github.com/ros-simulation/gazebo_ros_demos/blob/kinetic-devel/rrbot_desc
     <!-- ros_control plugin -->
     <gazebo>
       <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
-        <parameters>$(find ros2_control_demo_bringup)/config/rrbot_gazebo_forward_controller_position.yaml</parameters>
```
Comment (haider8645) [inline ros2_control_demo_description/rrbot_description/gazebo/rrbot.gazebo.xacro:13]: We do not need this file anymore and both gazebo and non-gazebo examples use the rrbot_controllers since it was intended to add the forward_position_controller to the gazebo example as well.

File: ros2_control_demo_bringup/config/rrbot_joint_trajectory_publisher.yaml (hunk 4 above)
Comment (haider8645) [reply ros2_control_demo_bringup/config/rrbot_joint_trajectory_publisher.yaml]: This file has been deleted and the gazebo example also uses the rrbot_controllers.yaml

File: README.md (hunk 1 above)
Comment (haider8645) [reply README.md]: Other examples also do not mention the controllers yaml or urdf. i do not understand what you mean. The gazebo example also uses the rrbot_controllers.yaml file like the non-gazebo example

Comment (destogl) [reply README.md]: Oh, I see now… you are working on foxy. Please make PR against master. Otherwise, we are correcting all the time some old code. We can then rebase this to foxy later.

File: README.md (hunk 2 above)
Comment (destogl) [reply README.md:301]: you are very welcome to do that, but please do PR against master.

File: ros2_control_demo_bringup/config/rrbot_gazebo_joint_trajectory_controller.yaml (hunk 3 above)
Comment (destogl) [reply ros2_control_demo_bringup/config/rrbot_gazebo_joint_trajectory_controller.yaml]: Yes, this is mostly the difference. Actually, difference in trajectory interpolation vs. directly sending commands.

File: README.md (hunk 2 above)
Comment (haider8645) [reply README.md]: @destogl Helloo, I updated everything for the master and changed the base to master. However, I guess because I was working on the foxy and I merged master into it there are those 3 commits from foxy branch(1 from you, 2 from bot) also now part of this PR. Do you know how I can easily remove them?

Comment (haider8645) [reply README.md]: The only I can think of to remove those commits is to switch to master, update it to latest state and cherry pick my commits one at a time. Is the a possible way forward?

File: README.md [hunk 6]
```diff
@@ -501,6 +521,9 @@ Available controllers:
 - `fts_broadcaster[force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster]`
 - `joint_state_broadcaster[joint_state_broadcaster/JointStateBroadcaster]`
 
+Notes:
+  - Wrench messages are not displayed properly in Rviz as NaN values are not handled in Rviz and FTS Broadcaster may send NaN values.
```
Comment (bmagyar) [inline README.md:525]: I believe we added a tickbox to the rviz plugin, once ticked, rviz will not freak out on NaN values

Comment (haider8645) [reply README.md]: Thanks for pointing that out. I have cleaned up the Readme so that only the comments related to this gazebo rrbot update are in it.

File: README.md (hunk 2 above)
Comment (haider8645) [reply README.md]: Turns out that those commits do not really appear in the File Changed. The file changed now look good for my pov.

Comment (haider8645) [reply README.md]: @destogl could we please merge this? It has been open for a long time now

**Comment by dcconner** [comment]
I'm curious if you can give an update on status of Gazebo and RRBot demos with ROS 2.  I find a collection of abandoned PR/issues such as this one, and notes that you're going to remove Gazebo reference from README #231.  I was hoping to use RRBot demo in a class I'm teaching as I did with ROS 1, but it seems that will be a dead end for the near term.  Can you shed some light on the issues preventing the old style demo from ROS 1 with Gazebo and tuning the PID values.  If I am understanding the current state of demos, they are just a kinematic demos using RViz.

**Comment by christophfroehlich** [comment]
> I'm curious if you can give an update on status of Gazebo and RRBot demos with ROS 2. I find a collection of abandoned PR/issues such as this one, and notes that you're going to remove Gazebo reference from README #231. I was hoping to use RRBot demo in a class I'm teaching as I did with ROS 1, but it seems that will be a dead end for the near term. Can you shed some light on the issues preventing the old style demo from ROS 1 with Gazebo and tuning the PID values. If I am understanding the current state of demos, they are just a kinematic demos using RViz.

We decided to include another example showing a typical Gazebo Classic + Gazebo setup, probably using the contributions from this PR. More complex examples should be maintained at/be added to https://github.com/ros-controls/gazebo_ros2_control and https://github.com/ros-controls/gz_ros2_control, respectively.

**Comment by christophfroehlich** [comment]
Merging this PR into #269
